### ISSUE TITLE
Demo cases?

### ISSUE THREAD

**Issue by BinhongLiu** [CongLabCode/DPAM#9]
Hi, I just test DPAM on my local Linux server using AF-A0A0K2WPR7-F1-model_v4 protein model from Caenorhabditis elegans. It seems to be executed well, but some warnings appeared. 

1. - 13:41:05.697 WARNING: Number of hits passing 2nd prefilter (reduced from 70733 to allowed maximum of 20000).
You can increase the allowed maximum using the -maxfilt <max> option.
2. - 13:47:09.756 WARNING: Number of hits passing 2nd prefilter (reduced from 87397 to allowed maximum of 20000).
You can increase the allowed maximum using the -maxfilt <max> option.
3. - 13:49:14.511 WARNING: Input alignment A0A0K2WPR7.hmm looks like aligned FASTA instead of A2M/A3M format. Consider using '-M first' or '-M 50'
4. cat: /home/liuhongbin/soft/DPAM-main/output_dir2/iterativeDali_A0A0K2WPR7/A0A0K2WPR7_*_hits: 没有那个文件或目录

the log file is attached here.
[log.txt](https://github.com/CongLabCode/DPAM/files/10810675/log.txt)

And the final number of found domains was 4, but in the model_organisms folder it's 3.
4 domains:
D1	31-65
D2	141-165
D3	166-260
D4	296-320
3 domains:
D1	6-65
D2	156-260
D3	261-335

Has the tool been run well? Should I fix these warnings? And I'm not sure the difference between two results was caused by the update of database or some other reasons.
Could you provide some demo AF2 models and results for the test?
Many thanks!

**Comment by jzhan6** [comment]
@BinhongLiu 
 Sorry for the late reply. It seems Dali does not run correctly in your case. So the pipeline only used hhsearch hits. I will put some demos in the following days. Sorry for the confusion.

**Comment by BinhongLiu** [comment]
I checked the environment and found that the Dali could be performed correctly. I'm not sure which step or tool caused the problem.

![微信图片_20230313110627](https://user-images.githubusercontent.com/35453307/224599190-816e25e2-c3af-4b13-89c8-897812fa549f.png)

the output folder
![微信图片_20230313110702](https://user-images.githubusercontent.com/35453307/224599211-78198d11-f540-4e77-be79-a2d2db128aa6.png)

**Comment by CongLabCode** [comment]
Sorry for the late reply.  Here is my result. You can see that your A0A0K2WPR7_iterativdDali_hits is empty, while mine is 120K. So somehow, your iterative Dali process does not run correctly in your system. I uploaded the demo for this protein in zip file on github. 
![image](https://user-images.githubusercontent.com/111030732/225179689-3f5d453e-b70c-4137-964d-0c1f7fb17627.png)

**Comment by BinhongLiu** [comment]
Yes. Was this _iterativdDali_hits file created by step 7 or step 6? Is there any method to debug the error? Thanks!
Best!

**Comment by CongLabCode** [comment]
step7. You have already successfully finished the first six steps, you can just run step 7 to see if any error messages are reported.

**Comment by BinhongLiu** [comment]
Yes.
I tested step 7 but nothing was reported. It seemed to be that nothing was done because the script was completed rapidly after I entered the command. 

`python step7_iterative_dali_aug_multi.py A0A0K2WPR7 20 output_dir2 /home/liuhongbin/database/DPAM`

![image](https://user-images.githubusercontent.com/35453307/225573779-be36496a-37ff-48e7-887d-a0b7e9e2641d.png)

Thank you for your help with this error!

**Comment by BinhongLiu** [comment]
Hi,
 Sorry to bother you again. Would you like to help me fix this error? Thanks!

**Comment by jzhan6** [comment]
Absolutely, I have requested a colleague in the lab to test my GitHub code
in their environment to ensure a smooth installation process. Additionally,
I discovered a bug in my previous download scripts, which failed to
download crucial files such as ECOD_norms, ecod.latest.domains,
ECOD_length, and ECOD_pdbmap. Could you please confirm if you have
downloaded these files? Meanwhile, please note that the step7 script
necessitates a data structure like {data_dir}/ECOD70/{edomain}.pdb. So you
have to put ECOD70 under the {data_dir} you provide to the wrapper.

**Comment by BinhongLiu** [comment]
Yes. I think I've prepared the database properly.
![image](https://user-images.githubusercontent.com/35453307/227864712-ac06cb34-8fee-492a-b291-3b64fff104da.png)
![image](https://user-images.githubusercontent.com/35453307/227864903-7de109bc-d6f7-4a4f-8f6c-8c3cab14a9ec.png)

**Comment by BinhongLiu** [comment]
Hi, I recently do some new tests, and it seems to be that the error happened at step 7.
I test step 7 using the files you've produced (https://github.com/CongLabCode/DPAM/blob/main/A0A0K2WPR7.zip)
`python /home/liuhongbin/soft/DPAM-main/step7_iterative_dali_aug_multi2.py A0A0K2WPR7 20 /home/liuhongbin/soft/DPAM-main/output_dir2 /home/liuhongbin/database/DPAM`
Nothing happened.

I checked the step7 script and it seems that the dali.pl is run iteratively to search against the ECOD70 database.
So I test the individual code with one pdb from ECOD70.
`dali.pl --pdbfile1 ../A0A0K2WPR7.pdb --pdbfile2 /home/liuhongbin/database/DPAM/ECOD70/000000003.pdb --dat1 ./ -dat2 ./ --outfmt summary,alignments,transrot >& log`

Here is what produced in the iterativeDali_A0A0K2WPR7 folder.
![image](https://user-images.githubusercontent.com/35453307/230923454-8b2861ed-4453-4042-a8f6-57edadf0301e.png)

Therefore, it seems to be that the dali.pl is also ok and could be run properly. 
 
I think these information may provide some ideas that what cause the error.

**Comment by jzhan6** [comment]
Hi Hongbin,

No. I think it should be deleted by the scripts. Here our scripts have very
intensive i/o usage. I do not know if personal computer handle it well.It
may be caused by Dali. Dali is very old software although it is still the
best for structure alignment and it has limited length allowed for the
input filename.   The absolute path (/home/liuhongbin/database/
DPAM/ECOD70/000000003.pdb) may be too long for it. Let me check if it is
true. Thanks for your details.

Best,
Jing Zhang
